### PR TITLE
Adding Binding -> Cuno-Onewire (Based on Transport-Cul)

### PR DESCRIPTION
**Pull request by querdenker2k**
Can be used to access Onewire Devices through the CUNO

### CODE REVIEW

**Comment by teichsta** [comment]
Hi @querdenker2k thanks for this contribution. Hope to find the time for the review soon. Best, Thomas E.-E.

File: bundles/binding/org.openhab.binding.onewire-cuno/pom.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		<deb.name>openhab-addon-binding-onewire-cuno</deb.name>
+		<deb.description>${project.name}</deb.description>
+	</properties>
+
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>org.openhab.binding</groupId>
+	<artifactId>org.openhab.binding.onewire-cuno</artifactId>
+
+	<packaging>eclipse-plugin</packaging>
+
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.onewire-cuno/pom.xml]: add the jdeb dependency (see other bindings how)

File: bundles/binding/org.openhab.binding.onewire-cuno/src/main/java/org/openhab/binding/onewire_cuno/OnewireCunoBindingProvider.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * @author Robert Delbrück
+ * @since 1.7.0
+ */
+public interface OnewireCunoBindingProvider extends BindingProvider {
+
+	public OnewireCunoBindingConfig getConfigForItemName(String itemName);
+
+	public OnewireCunoBindingConfig getConfigForAddress(String address);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.onewire-cuno/src/main/java/org/openhab/binding/onewire_cuno/OnewireCunoBindingProvider.java]: "public" keyword can be omitted
please add javadoc

File: bundles/binding/org.openhab.binding.onewire-cuno/src/main/java/org/openhab/binding/onewire_cuno/internal/OnewireCunoBinding.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+		return "Onewire (Cuno) Refresh Service";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void execute() {
+		// Nothing to do here
+	}
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.onewire-cuno/src/main/java/org/openhab/binding/onewire_cuno/internal/OnewireCunoBinding.java]: when there is no polling needed you should better extend AbstractBinding

File: bundles/binding/org.openhab.binding.onewire-cuno/src/main/java/org/openhab/binding/onewire_cuno/internal/OnewireCunoBinding.java [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		// nothing to do
+	}
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.onewire-cuno/src/main/java/org/openhab/binding/onewire_cuno/internal/OnewireCunoBinding.java]: remove this method

File: bundles/binding/org.openhab.binding.onewire-cuno/src/main/java/org/openhab/binding/onewire_cuno/internal/OnewireCunoGenericBindingProvider.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+							+ item.getName()
+							+ "' is of type '"
+							+ item.getClass().getSimpleName()
+							+ "', only NumberItems are allowed - please check your *.items configuration");
+		}
+
+		if (bindingConfig.length() != 4) {
+			throw new BindingConfigParseException(
+					"The configured address must consist of just 2 bytes housecode");
+		}
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.onewire-cuno/src/main/java/org/openhab/binding/onewire_cuno/internal/OnewireCunoGenericBindingProvider.java]: this method is only intended to validate the ItemType. Configuration validation should be executed in processBindingConfiguration()

**Comment by teichsta** [comment]
Hi @querdenker2k,

please find my review comments inline and some general notes below:

* update @version to 1.7.0
* update license header using "mvn -Dyear=2015 license:format”
* add javadoc on class level (containing @author @since) and for (at least) all public methods
* add your binding to distribution/pom.xml

Please also create the according Wiki page and refer to it here once it is ready. Finally please squash your commits into one.

Thanks again,

Thomas E.-E.

**Comment by querdenker2k** [comment]
https://github.com/openhab/openhab/wiki/Onewire-CUNO

**Comment by querdenker2k** [comment]
Hi teichsta,

I think i need help. I tried to squash my commits into one but the result is that now there are more commits then just one. Sorry for circumstances but i normally just using svn and hg...

**Comment by teichsta** [comment]
does this little how-to help? https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit. If not you could simply create a new branch. Best, Thomas E.-E.

**Comment by querdenker2k** [comment]
Hi teichsta,

I decided to create a new branch. Its all in one commit now and all remarks from you are resolved.

robert

**Comment by teichsta** [comment]
superseded by #2549

**Pull request closed without merging**